src/Message/MessageInterface.php

```diff
@@ -27,5 +27,12 @@ public function getHandlerName(): string;
      *
      * @return mixed
      */
-    public function getData();
+    public function getData(): mixed;
+
+    /**
+     * Returns some message metadata: timings, attempts count, metrics, etc.
```

Comment:
Are we sure we want to transmit this data in an array? Why not make them separate members of the message class?

getDelay(): float

getAttempts(): int

Reply:
Delay and attempts count are just single examples of potential project-specific metadata. We shouldn't force user to create `getDelay()` and `getAttempts()` methods when he doesn't use this metadata.